content/post/deeplearning_in_go_part_1.md

```diff
@@ -0,0 +1,393 @@
+
+This corresponds to this part of the cleaned up graph (with unnecessary nodes removed):
+
+<div style="margin-left:auto; margin-right:auto;">
+<svg style="width:100%; height:auto;"
+ viewBox="0.00 0.00 590.07 116.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 112)">
+<title>%3</title>
+<polygon fill="white" stroke="none" points="-4,4 -4,-112 586.075,-112 586.075,4 -4,4"/>
+<!-- args1 -->
+<g id="node1" class="node"><title>args1</title>
+<ellipse fill="none" stroke="black" cx="291.037" cy="-90" rx="97.4827" ry="18"/>
+<text text-anchor="middle" x="291.037" y="-86.3" font-family="Times,serif" font-size="14.00">*ast.BinaryExpr (Op: +)</text>
+</g>
+<!-- lhs -->
+<g id="node2" class="node"><title>lhs</title>
+<ellipse fill="none" stroke="black" cx="141.037" cy="-18" rx="141.075" ry="18"/>
+<text text-anchor="middle" x="141.037" y="-14.3" font-family="Times,serif" font-size="14.00">*ast.BasicLit (Kind: INT) (Value: 1)</text>
+</g>
+<!-- args1&#45;&gt;lhs -->
+<g id="edge1" class="edge"><title>args1&#45;&gt;lhs</title>
+<path fill="none" stroke="black" d="M257,-73.1159C235.88,-63.26 208.509,-50.4867 185.663,-39.825"/>
+<polygon fill="black" stroke="black" points="187.047,-36.6089 176.505,-35.5516 184.087,-42.9522 187.047,-36.6089"/>
+</g>
+<!-- rhs -->
+<g id="node3" class="node"><title>rhs</title>
+<ellipse fill="none" stroke="black" cx="441.037" cy="-18" rx="141.075" ry="18"/>
+<text text-anchor="middle" x="441.037" y="-14.3" font-family="Times,serif" font-size="14.00">*ast.BasicLit (Kind: INT) (Value: 1)</text>
+</g>
+<!-- args1&#45;&gt;rhs -->
+<g id="edge2" class="edge"><title>args1&#45;&gt;rhs</title>
+<path fill="none" stroke="black" d="M325.075,-73.1159C346.195,-63.26 373.566,-50.4867 396.412,-39.825"/>
+<polygon fill="black" stroke="black" points="397.988,-42.9522 405.57,-35.5516 395.028,-36.6089 397.988,-42.9522"/>
+</g>
+</g>
+</svg>
+
+</div>
+
+
+The values of the program flow from bottom to up. When the program runs, it starts right at the top. The arrows point to what each node depends on. So for example, the value of the `*ast.BinaryExpr` node is dependent on the values of `*ast.BasicLit (Kind: INT)`. Since we know both values are `1`, and we know what `+` does, we know that the value at the node `*ast.BinaryExpr` is `2`.
+
+## Equations As Graphs ##
+
+Why did you just sit through the above to know something you probably already know? Well, it's because deep learning is really in its core, just a bunch of mathematical equations. Wait, don't go yet! It's not that scary. I am personally of the opinion that one can't really do deep learning (or any machine learning, really) without understanding the mathematics behind it. And in my experience there hasn't been a better way to learn it than visually, if only to internalize the concepts.
```

Comment:
Maybe, "Now why did we spent all that time show `1+1` in graph form. Well, it's because ..."